src/frontend/js/modules/core.js

```diff
@@ -223,16 +224,46 @@ const _ = require('lodash');
     })
   /* global DOMPurify: false */
     .filter('esnDomPurify', function($sce) {
-      return function(dirty) {
-        DOMPurify.addHook('afterSanitizeAttributes', function(node) {
-        // set all elements owning target to target=_blank
-          if ('href' in node) {
-            node.setAttribute('target', '_blank');
-            node.setAttribute('rel', 'nofollow noopener');
+      function _computeCssRule(cssSelector) {
+        return '.mail_container ' + cssSelector.replaceAll(/([.#])/g, '$1x_');
+      }
+
+      function _addDivContainer(mailContent) {
+        return '<div class="mail_container">' + mailContent + '</div>';
+      }
+
+      const parser = new cssjs.cssjs();
+
+      DOMPurify.setConfig({
+        WHOLE_DOCUMENT: true,
+        ADD_ATTR: ['target'],
+        FORBID_TAGS: ['input', 'form']
+      });
+      DOMPurify.addHook('uponSanitizeElement', function(node, data) {
+        if (data.tagName === 'style') {
+          const parsedStyle = parser.parseCSS(node.textContent);
+
+          for (let i = 0; i < parsedStyle.length; i++) {
```

Comment:
```suggestion
          parsedStyle = parsedStyle.map(style => {
              style.selector = _computeCssRule(style.selector);
```

Reply:
The idea is good, though we should not use map in this precise case: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map#when_not_to_use_map
I will use map but change the code a little.

Reply:
you can still use ForEach